### ISSUE TITLE
False Context manager 'async_generator' doesn't implement __enter__ and __exit__

### ISSUE THREAD

**Issue by davetapley** [pylint-dev/pylint#10408]
### Bug description

```python
# repro.py

# inspired by:
# https://docs.python.org/3/library/contextlib.html#contextlib.asynccontextmanager

import time
from contextlib import asynccontextmanager


@asynccontextmanager
async def timeit():
    now = time.monotonic()
    try:
        yield
    finally:
        print(f'it took {time.monotonic() - now}s to run')


with timeit():
    print('hello world')
```

### Configuration

```ini

```

### Command used

```shell
pylint repro.py
```

### Pylint output

```python
repro.py:14:0: E1129: Context manager 'async_generator' doesn't implement __enter__ and __exit__. (not-context-manager)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.10
Python 3.12.2 (main, Feb 25 2024, 04:38:01) [Clang 17.0.6 ]
```

### Additional dependencies

```python

```

**Comment by cdce8p** [comment]
`timeit` is an async context manager so you'll need to use `async with`.
```py
async def main():
    async with timeit():
        print('hello world')
```

Technically the error message is correct since `@asynccontextmanger` only implements `__aenter__` and `__exit__`. Maybe a different error message would be more helpful here? Something like "timeit is an async context manager, you'll need to use async with".
